ext/ffi/tests/bug79571.phpt

```diff
@@ -0,0 +1,28 @@
+--TEST--
+Bug #79571 (FFI: var_dumping unions may segfault)
+--SKIPIF--
+<?php
+if (!extension_loaded('ffi')) die('skip ffi extension not available');
+?>
+--FILE--
+<?php
+$ffi = FFI::cdef(<<<EOF
+    typedef union {
+        int num;
+        char *str;
+    } my_union;
+EOF);
+
+$union = $ffi->new('my_union');
+$union->num = 42;
+var_dump($union);
+var_dump($union->num);
+?>
+--EXPECTF--
+object(FFI\CData:union <anonymous>)#%d (2) {
+  ["num"]=>
+  int(42)
+  ["str"]=>
+  int(42)
```

Comment:
I think maybe a pointer format here is better,  str => 0x28 ?

Reply:
I'm not happy treating this as `IS_LONG`, but I'm not sure that `IS_STRING` would be better. Actually, I think I'd like to have this as `IS_PTR`, but that would require adjustments to `php_var_dump()` etc.

Reply:
I see,  not sure, how about  get_debug handler?

Reply:
yeah, IS_PTR seems better.. anyway,  I think the current state is also okey, thanks